tests/page-objects/default/reports/reports.wdio.page.js

```diff
@@ -205,51 +220,44 @@ const isReportSelected = async (uuid) => {
   return await (await checkbox).isExisting();
 };
 
-const selectReports = async (uuids) => {
+const toggleSelectReport = async (selectReport, uuids, verifyFn, shouldHideElements = false) => {
   for (const uuid of uuids) {
-    if (!(await isReportSelected(uuid))) {
-      await (await reportCheckbox(uuid)).click();
+    const isSelected = await isReportSelected(uuid);
+    if ((selectReport && !isSelected) || (!selectReport && isSelected)) {
```

Comment:
I am not sure about this line, we might be adding some extra complexity, and maybe it is not that beneficial. 
What do you think @lorerod ? Do you think that the method is clear? would you change it?

Reply:
I left some suggestions to simplify it. If you are still uncomfortable with the code, we can leave it as it was separately; sometimes, duplicate code is better when it makes the code more easy to understand. In my opinion, it's not too complicated.